Makefile.one

```diff
@@ -0,0 +1,443 @@
+		-I $(KERN_BUILD_PATH)/arch/$(LINUX_ARCH)/include/generated/uapi \
+		-I $(KERN_SRC_PATH)/include \
+		-I $(KERN_BUILD_PATH)/include \
+		-I $(KERN_SRC_PATH)/include/uapi \
+		-I $(KERN_BUILD_PATH)/include/generated \
+		-I $(KERN_BUILD_PATH)/include/generated/uapi \
+		-I$(OUTPUT_DIR)/tracee.bpf \
+		-Wunused \
+		-Wall \
+		-Wno-frame-address \
+		-Wno-unused-value \
+		-Wno-unknown-warning-option \
+		-Wno-pragma-once-outside-header \
+		-Wno-pointer-sign \
+		-Wno-gnu-variable-sized-type-not-at-end \
+		-Wno-deprecated-declarations \
+		-Wno-compare-distinct-pointer-types \
+		-Wno-address-of-packed-member \
+		-fno-stack-protector \
+		-fno-jump-tables \
+		-fno-unwind-tables \
+		-fno-asynchronous-unwind-tables \
+		-xc -O2 -g -emit-llvm \
+		-c $(TRACEE_EBPF_OBJ_SRC) \
+		-o $(@:.o=.ll)
+	$(CMD_LLC) \
+		-march=bpf -mcpu=$(BPF_VCPU) \
+		-filetype=obj \
+		-o $@ \
+		$(@:.o=.ll)
+	$(CMD_RM) $(@:.o=.ll)
+
+.PHONY: clean-non-core
+clean-non-core:
+	$(CMD_RM) -rf $(OUTPUT_DIR)/tracee.bpf.$(NONCORE_TAG).o
+
+#
+# co-re ebpf
+#
+
+TRACEE_EBPF_OBJ_SRC = ./tracee-ebpf/tracee/tracee.bpf.c
+
+.PHONY: core
+core: $(OUTPUT_DIR)/tracee.bpf.core.o
+
+$(OUTPUT_DIR)/tracee.bpf.core.o: $(OUTPUT_DIR)/tracee.bpf $(TRACEE_EBPF_OBJ_SRC)
+	$(CMD_CLANG) \
+		-D__TARGET_ARCH_$(LINUX_ARCH) \
+		-D__BPF_TRACING__ \
+		-DCORE \
+		-I./tracee-ebpf/tracee/co-re/ \
+		-I$(OUTPUT_DIR)/tracee.bpf \
+		-target bpf \
+		-O2 -g \
+		-target bpf -march=bpf -mcpu=$(BPF_VCPU) \
+		-c $(TRACEE_EBPF_OBJ_SRC) \
+		-o $@
+
+.PHONY: clean-core
+clean-core:
+	$(CMD_RM) -rf $(OUTPUT_DIR)/tracee.bpf.core.o
+
+#
+# tracee-ebpf
+#
+
+STATIC ?= 0
+GO_TAGS_EBPF = core,ebpf
+CGO_EXT_LDFLAGS =
+
+ifeq ($(STATIC), 1)
+    CGO_EXT_LDFLAGS += -static
+    GO_TAGS_EBPF := $(GO_TAGS_EBPF),netgo
+endif
+
+CUSTOM_CGO_CFLAGS = "-I$(abspath $(OUTPUT_DIR)/libbpf)"
+CUSTOM_CGO_LDFLAGS = "-lelf -lz $(abspath $(OUTPUT_DIR)/libbpf/libbpf.a)"
+
+GO_ENV_EBPF =
+GO_ENV_EBPF += GOOS=linux
+GO_ENV_EBPF += CC=$(CMD_CLANG)
+GO_ENV_EBPF += GOARCH=$(GO_ARCH)
+GO_ENV_EBPF += CGO_CFLAGS=$(CUSTOM_CGO_CFLAGS)
+GO_ENV_EBPF += CGO_LDFLAGS=$(CUSTOM_CGO_LDFLAGS)
+
+TRACEE_EBPF_SRC=$(shell find ./tracee-ebpf/ -name '*.go' ! -name '*_test.go')
+
+.PHONY: tracee-ebpf
+tracee-ebpf: $(OUTPUT_DIR)/tracee-ebpf
+
+$(OUTPUT_DIR)/tracee-ebpf: $(OUTPUT_DIR)/tracee.bpf.core.o $(TRACEE_EBPF_SRC)
+	$(GO_ENV_EBPF) $(CMD_GO) build \
+		-tags $(GO_TAGS_EBPF) \
+		-ldflags "-w \
+			-extldflags \"$(CGO_EXT_LDFLAGS)\" \
+			-X main.version=\"$(VERSION)\" \
+			" \
+		-v -o $@ ./tracee-ebpf
+
+.PHONY: clean-tracee-ebpf
+clean-tracee-ebpf:
+	$(CMD_RM) -rf $(OUTPUT_DIR)/tracee-ebpf
+
+.PHONY: test-tracee-ebpf
+test-tracee-ebpf: tracee-ebpf
+	$(GO_ENV_EBPF) $(CMD_GO) test \
+		-tags $(GO_TAGS_EBPF) \
+		-v ./tracee-ebpf/...
+
+#
+# tracee-rules
+#
+
+GO_ENV_RULES =
+GO_ENV_RULES += GOOS=linux
+GO_ENV_RULES += CC=$(CMD_CLANG)
+GO_ENV_RULES += GOARCH=$(GO_ARCH)
+
+TRACEE_RULES_SRC=$(shell find ./tracee-rules/ \
```

Comment:
can we make this similar to the ebpf source finder: `find ./tracee-ebpf/ -name '*.go' ! -name '*_test.go'`